File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> indices;
        map<int,int> complements;//complemnt:index
        for(int i=0;i<nums.size();i++){
            if(complements.find(nums[i])!=complements.end())//it exists
            {
                indices.push_back(complements[nums[i]]);
                indices.push_back(i);
                return indices;
            }
            else{//not found
                complements[target-nums[i]]=i;
            }
            }
       
        
        return indices; 
        }
    };

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans=[]
        dict={}
        for idx,num in enumerate(nums):
            if target-num not in dict:
                dict[num]=idx
            else:
                ans.append(dict[target-num])
                ans.append(idx)
                return ans
        